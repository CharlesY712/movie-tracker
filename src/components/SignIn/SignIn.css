.sign-in-form {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-flow: row wrap;
  background-color: var(--card-bg);
  padding: 10px;
  border-radius: 10px;
  border: 1px solid var(--card-border);
  position: fixed;
  margin: 5px;
  min-height: 100px;
  min-width: 145px;
  text-align: center;
  width: 10%;
}
.sign-in-form p {
  margin: 5px;
}
.submit-btn {
  min-width: 150px;
  width: 100%;
}
.pre-sign-in {
  box-sizing: border-box;
  width: 100%;
}
.post-sign-in {
  box-sizing: border-box;
  display: none;
  width: 100%;
}
