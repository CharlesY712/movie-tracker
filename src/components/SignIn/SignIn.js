import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { signIn } from '../../helper/apiCall';
import PropTypes from 'prop-types';


class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      errorMessage: ''
    };
  }

  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.user.length ? this.handleSignOut() : this.handleSignIn(event);
  }

  handleSignIn = async (event) => {
    const email = this.state.email.toLowerCase();
    const password = this.state.password;
    const user = await signIn(event, email, password);
    user.status ?
      this.handleUser(user.data)
      :
      this.handleError(user.error);

    this.setState({
      email: '',
      password: ''
    });
  }

  handleUser = (user) => {
    this.props.handleUser(user);
    this.props.history.push('/');
  }

  handleSignOut = () => {
    this.props.removeUser();
  }

  handleError = (error) => {
    this.setState({errorMessage: error.message});
    alert('Email and Password do not match!');
  }

  createAccount = async () => {

  }

  render() {
    const { email, password } = this.state;
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input
            type="email"
            name="email"
            value={email}
            placeholder="Enter Email"
            onChange={this.handleChange}
          />
          <input
            type="text"
            name="password"
            value={password}
            placeholder="Enter Your Password"
            onChange={this.handleChange}
          />
          <button type="Submit">{this.props.user.length ?
            "Sign Out":
            "Sign In"}</button><br/>
          <p>tman2272@aol.com password</p>
        </form>
        <NavLink to="/account">Create an Account</NavLink>
      </div>
    );
  }
}

SignIn.propTypes = {
  user: PropTypes.object,
  handleUser: PropTypes.func,
  history: PropTypes.object,
  removeUser: PropTypes.func
};

export default SignIn;
