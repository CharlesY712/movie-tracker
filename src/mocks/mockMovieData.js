module.exports = {results: [{
adult:false,
id:337167,
overview: "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
popularity:572.203541,
poster_path:"/jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
release_date:"2018-02-07",
title:"Fifty Shades Freed",
video:false,
vote_average:6.1,
vote_count:1097
},
{
adult:false,
id:284054,
overview:"King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan special forces), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
popularity:329.754985,
poster_path:"/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
release_date:"2018-02-13",
title:"Black Panther",
video:false,
vote_average:7.4,
vote_count:3477
}]}
