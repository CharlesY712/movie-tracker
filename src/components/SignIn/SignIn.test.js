import React from 'react';
import { shallow } from 'enzyme';
import SignIn from './SignIn';
import { cleanData, cleanMovie } from '../../mocks/mockMovieData';

describe('SignIn' ,() => {
  let wrapper, user, history, mockHandleUser, mockRemoveuser;

  beforeEach(() => {
    user = [];
    // user = [{
    //   id: 1,
    //   name: "Bob",
    //   password: "ilovemywife",
    //   email: "billybob@gmail.com",
    //   favorites: cleanData
    // }];
    mockHandleUser = jest.fn();
    mockRemoveuser = jest.fn()
    history = { location: {pathname:'/signin'}}
    wrapper = shallow(
      <SignIn
        user={user}
        history={history}
        handleUser={mockHandleUser}
        removeUser={mockRemoveuser}
      />
    )
  });

  it('Should match the snapshout', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have a defualt state of email as a empty string', () => {

    expect(wrapper.state('email')).toEqual('');
  });

  it('Should have a defualt password set to a empty string', () => {

    expect(wrapper.state('password')).toEqual('');
  });

  it('Should have a default errorMessage to a empty string', () => {
    expect(wrapper.state('errorMessage')).toEqual('');
  });

  it('Should add the event to state when the input changes',() => {

    const event = { target: {
      name: 'email',
      value: 'bob@gmail.com'
    }};

    expect(wrapper.state('email')).toEqual('');
    wrapper.instance().handleChange(event);
    expect(wrapper.state('email')).toEqual('bob@gmail.com')
  })

  it.skip('Should call handleSignIn if ther is not a user', async() => {
    const event = { preventDefault: jest.fn()}
    wrapper.instance().handleSubmit(event);
    const expected = await wrapper.instance().handleSignIn()
    expect(expected).toHaveBeenCalledWith(event)
  })


})
