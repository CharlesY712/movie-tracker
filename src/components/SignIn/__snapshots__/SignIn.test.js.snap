// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Should match the snapshout 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    handleUser={[Function]}
    history={
        Object {
            "location": Object {
              "pathname": "/signin",
            },
            "push": [Function],
          }
    }
    removeUser={[Function]}
    user={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Email"
                    type="email"
                    value=""
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter Your Password"
                    type="text"
                    value=""
          />
          <button
                    type="Submit"
          >
                    Sign In
          </button>
          <br />
          <p>
                    tman2272@aol.com password
          </p>
</form>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/account"
>
          Create an Account
</NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              onChange={[Function]}
              placeholder="Enter Email"
              type="email"
              value=""
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Enter Your Password"
              type="text"
              value=""
/>,
            <button
              type="Submit"
>
              Sign In
</button>,
            <br />,
            <p>
              tman2272@aol.com password
</p>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Enter Email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Enter Your Password",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "type": "Submit",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "tman2272@aol.com password",
            },
            "ref": null,
            "rendered": "tman2272@aol.com password",
            "type": "p",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Create an Account",
          "to": "/account",
        },
        "ref": null,
        "rendered": "Create an Account",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter Email"
                        type="email"
                        value=""
            />
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter Your Password"
                        type="text"
                        value=""
            />
            <button
                        type="Submit"
            >
                        Sign In
            </button>
            <br />
            <p>
                        tman2272@aol.com password
            </p>
</form>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/account"
>
            Create an Account
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="email"
                onChange={[Function]}
                placeholder="Enter Email"
                type="email"
                value=""
/>,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Enter Your Password"
                type="text"
                value=""
/>,
              <button
                type="Submit"
>
                Sign In
</button>,
              <br />,
              <p>
                tman2272@aol.com password
</p>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Enter Email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Enter Your Password",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "type": "Submit",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "tman2272@aol.com password",
              },
              "ref": null,
              "rendered": "tman2272@aol.com password",
              "type": "p",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Create an Account",
            "to": "/account",
          },
          "ref": null,
          "rendered": "Create an Account",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
